client/src/pages/GrantAccessForm.js

```diff
@@ -0,0 +1,200 @@
+import { useState } from "react";
+import "./GrantAccessForm.css";
+import { post } from "../TenderClient";
+import Logo from "../assets/images/CTY-logo-rectangle.png";
+
+const GrantAccessForm = () => {
+	const [role, setRole] = useState("bidder");
+	const [buyerDetails, setBuyerDetails] = useState({
+		userType: "buyer",
+		company: "",
+		description: "",
+		address: "",
+		email: "",
+	});
+
+	const [bidderDetails, setBidderDetails] = useState({
+		userType: "bidder",
+		firstName: "",
+		lastName: "",
+		email: "",
+	});
+
+	const [resgisterStatus, setRegisterStatus] = useState("");
+
+	async function postDetails(endpoint, data) {
+		try {
+			const res = await post(`${endpoint}`, data);
+			if (res) {
+				setRegisterStatus("Successfully registered.");
+			} else {
+				setRegisterStatus("Registration failed!");
+			}
+		} catch (error) {
+			setRegisterStatus(error.message);
```

Comment:
This doesn't work, please test your code before sending it for review

Reply:
I have tested. Its showing the error message from server.

Reply:
Can you please share a screenshot of the error messages being shown to the user when the validation fails?

Reply:
![Screenshot from 2024-08-04 16-13-24](https://github.com/user-attachments/assets/4da68405-872a-4f96-bd76-3540a432f68d)

Reply:
That's not a validation error, that's a server error